src/components/MCQBlock.tsx

```diff
@@ -1,5 +1,175 @@
-import React from 'react'
+import {
+  Box,
+  Button,
+  Checkbox,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  FormControl,
+  FormControlLabel,
+  Paper,
+  Radio,
+  RadioGroup,
+  Typography,
+  styled,
+} from '@mui/material'
+import React, { useState } from 'react'
 
-export default function MCQBlock() {
-  return <div>MCQBlock</div>
+interface MCQBlockProps {
+  question: string
+  answerOptions: string[]
+  correctOptions: string[]
+}
+
+const ShowAnswerBtn = styled(Button)({
+  border: '1px solid #776E6E',
+  color: '#776E6E',
+  backgroundColor: 'transparent',
+  padding: '8px',
+  '&:hover': {
+    backgroundColor: '#8E8D8D',
+    borderColor: '#8E8D8D',
+    color: '#000000',
+  },
+  '&:active': {
+    backgroundColor: '#8E8D8D',
+    borderColor: '#8E8D8D',
+  },
+})
+
+export default function MCQBlock({
+  question,
+  answerOptions,
+  correctOptions,
+}: MCQBlockProps) {
+  const isMultiple = correctOptions.length > 1
+  const [selectedAnswers, setSelectedAnswers] = useState<string[]>([])
+
+  const [dialog, setDialog] = useState(false)
+  const [dialogText, setDialogText] = useState('')
+
+  const handleDialogClose = () => {
+    setDialog(false)
+  }
+
+  const handleAnswerSelection = (option: string) => {
+    // console.log("Selected ", {option});
+    if (selectedAnswers.includes(option)) {
+      setSelectedAnswers(
+        selectedAnswers.filter((selectedOption) => selectedOption !== option),
+      )
+    } else {
+      setSelectedAnswers([...selectedAnswers, option])
+    }
+  }
+
+  const checkAnswers = () => {
+    // console.log("Submit");
+    // console.log({selectedAnswers});
+    const isCorrect = [...correctOptions].every((option) =>
+      selectedAnswers.includes(option),
+    )
+    const text = isCorrect ? 'Correct!' : 'Incorrect! Try Again.'
+    // if (isCorrect) {
+    //   alert("Correct!");
+    // } else {
+    //   alert("Incorrect!");
+    // }
+    setDialog(true)
+    setDialogText(text)
+    // HAVE TO UPDATE HOW TO HANDLE AFTER SUBMIT
```

Comment:
can you format this like `TODO(Sath): Have to update how to handle after submit`

this way you can search the codebase for these comments easily (and there's ownership)